signac/__main__.py

```diff
@@ -342,26 +343,33 @@ def main_find(args):
     project = get_project()
 
     if args.show:
-        len_id = max(6, project.min_len_unique_id())
-
-        def format_lines(cat, _id, s):
-            if args.one_line:
-                if isinstance(s, dict):
-                    s = json.dumps(s, sort_keys=True)
-                return _id[:len_id] + ' ' + cat + '\t' + s
-            else:
-                return pformat(s, depth=args.show)
+        if args.sp:
+            warnings.warn(
+                "Specifying --show and --sp results in --sp being ignored")
```

Comment:
The `warnings` module is not good to show warnings like those, because it comes with a traceback. It would be better to either use the `logging` module or simply `print()`.